jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpGenerator.java

```diff
@@ -627,23 +628,36 @@ else if (contentLength != field.getLongValue())
 
                         case CONNECTION:
                         {
-                            boolean keepAlive = field.contains(HttpHeaderValue.KEEP_ALIVE.asString());
-                            if (keepAlive && _info.getHttpVersion() == HttpVersion.HTTP_1_0 && _persistent == null)
+                            boolean hasValue = StringUtil.isNotBlank(field.getValue());
+                            boolean hasKeepAlive = field.contains(HttpHeaderValue.KEEP_ALIVE.asString());
+                            boolean hasClose = field.contains(HttpHeaderValue.CLOSE.asString());
+
+                            if (http10 && hasKeepAlive)
                             {
-                                _persistent = true;
+                                // set persistence if unset
+                                if (_persistent == null)
+                                    _persistent = true;
                             }
-                            if (field.contains(HttpHeaderValue.CLOSE.asString()))
+                            else if (hasClose)
```

Comment:
I'm not sure we want the else here.  If it came `Connection: keep-alive, close` then either that is a 400 or at least we should close.     So perhaps invert the order of the tests and look for close, else look for keepAlive.

Reply:
I was reading the Hop-by-hop parts of the spec in light of other tests with have for this.

Looks like if we receive a comma delimited list, then it's a 400 Bad Request in normal operations.
BUT if we receive that when we are a proxy, then we have to act on all of the various Hop-by-hop headers (of which `Connection` is just one of them).

BUT we can create a response with multi-value `Connection` at all times (normal operation, and proxy operation), and we are only on the hook to act on a persistence state change for our hop.  But I'm at a loss on the exact meaning of "our hop" in this case.  If the request was `keep-alive` and response from the servlet is `close, keep-alive` vs `keep-alive, close`, what do we do at the prepareRequest / httpGenerator?

Reply:
I don't think we need to 400 for both a Keep-Alive and Close.  The close is a must-not persist and the Keep-Alive is a may persist, so they combine easily to a must-not persist.